packages/cli/src/ceramic-daemon.ts

```diff
@@ -341,33 +341,35 @@ export class CeramicDaemon {
     const documentsRouter = ErrorHandlingRouter(this.diagnosticsLogger)
     const multiqueriesRouter = ErrorHandlingRouter(this.diagnosticsLogger)
     const nodeRouter = ErrorHandlingRouter(this.diagnosticsLogger)
-    const pinsRouter = ErrorHandlingRouter(this.diagnosticsLogger)
     const recordsRouter = ErrorHandlingRouter(this.diagnosticsLogger)
     const streamsRouter = ErrorHandlingRouter(this.diagnosticsLogger)
     const collectionRouter = ErrorHandlingRouter(this.diagnosticsLogger)
     const adminCodesRouter = ErrorHandlingRouter(this.diagnosticsLogger)
     const adminModelRouter = ErrorHandlingRouter(this.diagnosticsLogger)
+    const adminPinsRouter = ErrorHandlingRouter(this.diagnosticsLogger)
     const adminNodeStatusRouter = ErrorHandlingRouter(this.diagnosticsLogger)
 
     app.use('/api/v0', baseRouter)
     baseRouter.use('/commits', commitsRouter)
     baseRouter.use('/documents', documentsRouter)
     baseRouter.use('/multiqueries', multiqueriesRouter)
     baseRouter.use('/node', nodeRouter)
-    baseRouter.use('/pins', pinsRouter)
     baseRouter.use('/records', recordsRouter)
     baseRouter.use('/streams', streamsRouter)
     baseRouter.use('/collection', collectionRouter)
     baseRouter.use('/admin/getCode', adminCodesRouter)
     baseRouter.use('/admin/models', adminModelRouter)
     baseRouter.use('/admin/status', adminNodeStatusRouter)
+    // Admin Pins Validate JWS Middleware
```

Comment:
what does this mean?  Also what does it do when you call `.use` on a Router twice with the same path?

Should the other Admin API endpoints use this same technique?  Why is this endpoint doing something different than the other ones?

Reply:
yeah its a common expressjs pattern - https://expressjs.com/en/guide/using-middleware.html, and you can register multiple and as mentioned below it uses next()

likely would make sense to use for all admin api validation, so that you can pull that verification out of core functions 

there is a could handful of things duplicated a bit here, I mostly tried to minimize changes to the existing adminapi since im less familiar with the codebase here now and shorter timeline, but these things could be revisited